.ci/test-project/packages/magictestlib/package.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import glob
+import os
+import shutil
+import socket
+from os import environ as env
+
+import llnl.util.tty as tty
+
+from spack import *
+
+
+def cmake_cache_entry(name, value, vtype=None):
+    """
+    Helper that creates CMake cache entry strings used in
+    'host-config' files.
+    """
+    if vtype is None:
+        if value == "ON" or value == "OFF":
+            vtype = "BOOL"
+        else:
+            vtype = "PATH"
+    return 'set({0} "{1}" CACHE {2} "")\n\n'.format(name, value, vtype)
+
+
+class Magictestlib(Package):
```

Comment:
Should this extend from `CachedCMakePackage` ? 
Or were you going for simplicity?

Reply:
I don't want to build cmake. Just want to test the basic mechanics.